ami/graph_nodes.py

```diff
@@ -366,10 +366,17 @@ def __call__(self, *args, **kwargs):
             self.res = np.roll(self.res, -nelem)
             self.res[..., -nelem:] = [args] if dims else args
         else:
-            if self.is_expanded:
-                self.res.extend(args)
-                self.idx = min(self.idx + len(args), self.N)
-            else:
+            if self.is_expanded: # this case is for collectors: args = (count, buffer)
+                # Logic to prevent self.res have a memory footprint > N
+                if args[0] + len(self.res) < self.N:
+                    self.res.extend(args[1])
+                else:
+                    # remove exactly enough to have a list of size N after addition of the new data
+                    remove = len(self.res) + args[0] - self.N
+                    self.res[:remove] = []
+                    self.res.extend(args[1])
+                self.idx = min(len(self.res), self.N)
+            else: # this case is for workers:  args = data
```

Comment:
These changes seem to break the ScalarPlot and ScatterPlot. Also the regression test is failing. Havent looked more into why yet.

Reply:
Good catch, I should have been more careful with the other box using this!
I am sure that the changes I made are going in the right direction, and might address the inconsistent number of points we see in those plots too!
Let me fix this!

Reply:
I think we should just have all GlobalTransformations return a count and value so that they are all consistent/interchangeable?

Reply:
I was thinking the same. That's in fact the initial approach I took, and what broke the test for the scalar plots etc.
The count is trivial for the rolling buffer, but consistency is probably a good idea